connector/connector_oidc.go

```diff
@@ -37,6 +40,37 @@ func (cfg *OIDCConnectorConfig) ConnectorType() string {
 	return OIDCConnectorType
 }
 
+type validateIdentityFunc func(id *oidc.Identity) error
+
+func validateIdentityPass(id *oidc.Identity) error {
+	return nil
+}
+
+func validateIdentityDomainFunc(want string) validateIdentityFunc {
+	return func(ident *oidc.Identity) error {
+		got, err := parseEmailDomain(ident.Email)
+		if err != nil {
+			return err
+		} else if want != got {
+			return errors.New("domain mismatch")
+		}
+		return nil
+	}
+}
+
+func parseEmailDomain(email string) (string, error) {
```

Comment:
I don't like that I had to write this function, but this is all I could find in stdlib: https://golang.org/pkg/net/mail/#ParseAddress